main/modes/games/2048/mode_2048.c

```diff
@@ -232,30 +234,95 @@ static void t48MainLoop(int64_t elapsedUs)
             // Take accel input if cells aren't moving
             if (t48->tiltControls && !t48->cellsAnimating)
             {
-                int16_t ox, oy, oz;
-                if (ESP_OK == accelGetOrientVec(&ox, &oy, &oz))
+                // int16_t ox, oy, oz;
+                float q[4];
+                esp_err_t aq = accelGetQuaternion(q);
+                if (ESP_OK == aq)
                 {
-                    if (ABS(ox) > ABS(oy))
+                    float qRelativeRotation[4];
+
+                    // this takes about 36 flops.
+                    mathComputeQuaternionDeltaBetweenQuaternions(qRelativeRotation, t48->quatBase, q);
+
+                    // qRelativeRotation represents the rotation from the initial state to now.
+                    // The quaternion is:  [w, x, y, z]
+                    //  the w term, more like the part that sums everything to a unit quaternion.
+                    //  the x term is the amount of rotation around the x axis.
+                    //  the y term is the amount of rotation around the y axis.
+                    //  the z term is the amount of rotation around the z axis.
+                    // Not quite like euler angles, but kinda?
+                    //
+                    // I am not sure why our coordinate frame is different, but, we can go back to
+                    // what the rest of the game logic is expecting here.
+                    int oy = qRelativeRotation[1] * 512;
+                    int ox = -qRelativeRotation[2] * 512;
+
+                    const int trigger = 104;
+                    const int release = 80;
+
+                    bool bXPressed      = ABS(ox) > trigger;
+                    bool bXReleased     = ABS(ox) < release;
```

Comment:
Interesting, a slightly different take than how I did it. I put this up to a channel vote. I don't think either is objectively better.